chain/transaction.go

```diff
@@ -372,6 +372,10 @@ func (t *Transaction) Execute(
 	}, nil
 }
 
+func (t *Transaction) GetSponsor() codec.Address {
```

Comment:
it's weird that we have both "GetSponsor" and "Sponsor".

Reply:
This is because `Sponsor` implements the `mempool.Item` interface while `GetSponsor` implements the `dsmr.Tx` interface. We could either wrap types to avoid this or rename both functions to be the same.

Reply:
We should rename both functions to be the same imo